### PR TITLE
HealthMate Ballistic Goggles

### PR DESCRIPTION
**Pull request by Sulaboy**
## About The Pull Request

Adds Ballistic Goggles with Medical HUD capabilities to the game as an alternative to the standard medical HUD available through the medic vendor as a point buy item purchasable for four vendor supply points. 

There are no mechanical changes from the standard HUD besides the ballistic goggles armor, which to my knowledge has no in game impact. A prescription version is also available for the one medic with bad vision who cannot wear glasses due to his medical HUD not catering to the visually disabled. 

The HUD can be toggled on and off with corresponding in-hand/icon, and worn sprites. The goggles are only equip-able to the glasses slot unlike standard ballistic goggles to prevent any possible issues from having multiple HUD items worn (Despite HUD items not providing benefit in the mask slot)

## Why It's Good For The Game

Looks like something a combat medic would wear instead of a dumb monocle.
Goggles Work, front

## Changelog
:cl: Sulaboy
add: Added marine ballistic healthmate goggles and prescription variant as medic point buy item.
imageadd: Added corresponding worn/inventory images for above added items
/:cl:

### CODE REVIEW

**Comment by Sulaboy** [comment]
Remove code improvement tag please, I have not improved anything because I suck.

**Comment by SilencedMP5A5** [comment]
based

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -455,7 +455,9 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 	if(istype(glasses,/obj/item/clothing/glasses/mgoggles))
 		overlays_standing[GOGGLES_LAYER] = glasses.make_worn_icon(body_type = species.name, slot_name = slot_glasses_str, default_icon = 'icons/mob/eyes.dmi', default_layer = GOGGLES_LAYER)
 		apply_overlay(GOGGLES_LAYER)
-	
+	if(istype(glasses,/obj/item/clothing/glasses/hud/medgoggles))
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/human/update_icons.dm:458]: Turn this istype into a flag please

File: code/modules/clothing/glasses/hud.dm [hunk 2]
```diff
@@ -68,6 +68,25 @@
 	hud_type = DATA_HUD_MEDICAL_ADVANCED
 	actions_types = list(/datum/action/item_action/toggle)
 
+/obj/item/clothing/glasses/hud/medgoggles
+	name = "marine ballistic healthmate goggles"
+	desc = "Standard issue TGMC goggles. With internal HealthMate HUD projector."
+	icon_state = "medgoggles"
+	item_state = "medgoggles"
+	deactive_state = "degoggles_medgoggles"
+	toggleable = 1
```
Comment (TiviPlus) [inline code/modules/clothing/glasses/hud.dm:77]: TRUE

File: code/modules/clothing/glasses/hud.dm [hunk 3]
```diff
@@ -68,6 +68,25 @@
 	hud_type = DATA_HUD_MEDICAL_ADVANCED
 	actions_types = list(/datum/action/item_action/toggle)
 
+/obj/item/clothing/glasses/hud/medgoggles
+	name = "marine ballistic healthmate goggles"
```
Comment (XSlayer300) [inline code/modules/clothing/glasses/hud.dm:72]:
```suggestion
	name = "\improper HealthMate ballistic goggles"
```

File: code/modules/clothing/glasses/hud.dm [hunk 4]
```diff
@@ -68,6 +68,25 @@
...
+	desc = "Standard issue TGMC goggles. With internal HealthMate HUD projector."
```
Comment (XSlayer300) [inline code/modules/clothing/glasses/hud.dm:73]:
```suggestion
	desc = "Standard issue TGMC goggles. With an internal HealthMate HUD projector."
```

File: code/modules/clothing/glasses/hud.dm [hunk 5]
```diff
@@ -68,6 +68,25 @@
...
+	deactive_state = "degoggles_medgoggles"
+	toggleable = 1
+	hud_type = DATA_HUD_MEDICAL_ADVANCED
+	actions_types = list(/datum/action/item_action/toggle)
+	soft_armor = list("melee" = 40, "bullet" = 40, "laser" = 0, "energy" = 15, "bomb" = 35, "bio" = 10, "rad" = 10, "fire" = 30, "acid" = 30)
+	flags_equip_slot = ITEM_SLOT_EYES
+
+/obj/item/clothing/glasses/hud/medgoggles/prescription
+	name = "prescription marine ballistic goggles"
+	desc = "Standard issue TGMC goggles. With internal HealthMate HUD projector. Contains prescription lenses."
```
Comment (XSlayer300) [inline code/modules/clothing/glasses/hud.dm:85]:
```suggestion
	desc = "Standard issue TGMC goggles. With an internal HealthMate HUD projector and prescription lenses."
```

File: code/modules/clothing/glasses/hud.dm [hunk 6]
```diff
@@ -68,6 +68,25 @@
...
+	item_state = "medgoggles"
+	deactive_state = "degoggles_medgoggles"
+	toggleable = 1
+	hud_type = DATA_HUD_MEDICAL_ADVANCED
+	actions_types = list(/datum/action/item_action/toggle)
+	soft_armor = list("melee" = 40, "bullet" = 40, "laser" = 0, "energy" = 15, "bomb" = 35, "bio" = 10, "rad" = 10, "fire" = 30, "acid" = 30)
+	flags_equip_slot = ITEM_SLOT_EYES
+
+/obj/item/clothing/glasses/hud/medgoggles/prescription
+	name = "prescription marine ballistic goggles"
```
Comment (XSlayer300) [inline code/modules/clothing/glasses/hud.dm:84]:
```suggestion
	name = "\improper HealthMate prescription ballistic goggles"
```

File: code/modules/mob/living/carbon/human/update_icons.dm (hunk 1 above)
Comment (Sulaboy) [reply code/modules/mob/living/carbon/human/update_icons.dm:458]: I don't know what I am doing and could not figure out how to make a flag.

File: code/modules/clothing/glasses/hud.dm (hunk 2 above)
Comment (Sulaboy) [reply code/modules/clothing/glasses/hud.dm:77]: I do not know what this means.

**Comment by Sulaboy** [comment]
I have no idea what I am doing, this is going in the trash.

**Comment by Sulaboy** [comment]
Can't figure out how to add my changes as I probably fucked something up on my github desktop. Making a new pull request instead.

**Pull request closed without merging**